src/java_bytecode/java_static_initializers.cpp

```diff
@@ -218,6 +248,157 @@ void create_static_initializer_wrappers(symbol_tablet &symbol_table)
   {
     const irep_idt &class_identifier = class_graph[node].class_identifier;
     if(needs_clinit_wrapper(class_identifier, symbol_table))
-      create_clinit_wrapper_symbols(class_identifier, symbol_table);
+    {
+      create_clinit_wrapper_symbols(
+        class_identifier, symbol_table, synthetic_methods);
+    }
+  }
+}
+
+/// Gets the class-id prefix from a symbol-id
+/// \param symbol_id: symbol identifier
+/// \return classs identifier, a prefix of symbol_id.
+static irep_idt static_field_id_to_class_id(const irep_idt &symbol_id)
```

Comment:
I'm not a fan of all this direct string manipulation, can we not look up the symbol in the symbol table and get the class from there somehow?

Reply:
Not for statics (either fields or functions) -- they're really just namespaced globals, and the only thing defining their namespace at the moment is the period-delimited name.

To do this properly we need to add an `annotations` member to `symbolt`. For some other stuff like this I've been tacking comments onto the symbol's type, but I'd rather not do that when it's not absolutely necessary, as such annotations are vulnerable to being accidentally erased when the type is updated.